ui/pages/confirmation/confirmation.js

```diff
@@ -258,14 +262,24 @@ export default function ConfirmationPage({
     return INPUT_STATE_CONFIRMATIONS.includes(type);
   };
 
-  const handleSubmit = () =>
-    templateState[pendingConfirmation.id]?.useWarningModal
+  const handleSubmit = async () => {
+    setLoading(true);
+    const submitResult = await (templateState[pendingConfirmation.id]
+      ?.useWarningModal
       ? setShowWarningModal(true)
       : templatedValues.onSubmit(
           hasInputState(pendingConfirmation.type)
             ? inputStates[MESSAGE_TYPE.SNAP_DIALOG_PROMPT]
             : null,
-        );
+        ));
+
+    if (submitResult?.error) {
+      setLoadingText(templatedValues.submitText);
+      setSubmitAlerts(submitResult.error);
+    } else {
+      setLoading(false);
```

Comment:
It looks like `submitResult.success` is no longer being used. Perhaps we could simplify this to just return the error, and drop the `success` boolean flag altogether?